tests/new_core/test_data_access.py

```diff
@@ -0,0 +1,399 @@
+"""
+Explicitly test the DataCatalog class and its methods.
+This module contains unit tests for the DataCatalog class, which is part of the climakitae.new_core.data_access.data_access module.
+These tests cover the initialization, default values, update methods, getting data, listing and printing clip boundaries, and resetting.
+"""
+
+import warnings
+from typing import Tuple
+from unittest.mock import MagicMock, Mock, PropertyMock, patch
+
+import geopandas as gpd
+import pandas as pd
+import pytest
+import xarray as xr
+
+from climakitae.core.paths import (
+    BOUNDARY_CATALOG_URL,
+    DATA_CATALOG_URL,
+    RENEWABLES_CATALOG_URL,
+)
+from climakitae.new_core.data_access.data_access import (
+    CATALOG_BOUNDARY,
+    CATALOG_CADCAT,
+    CATALOG_REN_ENERGY_GEN,
+    UNSET,
+    DataCatalog,
+    _get_closest_options,
+)
+
+
+def make_mock_objects() -> Tuple[MagicMock, MagicMock, MagicMock, pd.DataFrame]:
+    """Create mock objects for testing DataCatalog."""
+    # ESM catalog
+    mock_esm_catalog = MagicMock()
+    mock_esm_catalog.df = pd.DataFrame(
+        {"id": [1, 2, 3], "name": ["Alice", "Bob", "Charlie"], "age": [25, 30, 35]}
+    )
+
+    mock_subset = MagicMock()
+    mock_esm_catalog.search.return_value = mock_subset
+
+    ds1 = xr.Dataset(
+        data_vars={"tasmax": (["time"], [1.0, 2.0])},
+        coords={"time": [0, 1]},
+        attrs={"source_id": "MODEL1", "experiment_id": "historical"},
+    )
+    ds2 = xr.Dataset(
+        data_vars={"tasmax": (["time"], [3.0, 4.0])},
+        coords={"time": [0, 1]},
+        attrs={"source_id": "MODEL2", "experiment_id": "historical"},
+    )
+    mock_subset.to_dataset_dict.return_value = {"dataset1": ds1, "dataset2": ds2}
+
+    # Boundary catalog
+    mock_boundary_catalog = MagicMock()
+    mock_boundaries = MagicMock()
+    boundary_dict = {
+        "none": {"entire domain": 0},
+        "states": {"NV": 28, "CA": 16},
+        "CA watersheds": {
+            "Antelope-Fremont Valleys": 11,
+            "Aliso-San Onofre": 128,
+        },
+        "lat/lon": {"coordinate selection": 0},
+        "CA counties": {"Alameda County": 54, "Alpine County": 25},
+        "CA Electric Load Serving Entities (IOU & POU)": {
+            "San Diego Gas & Electric": 3,
+            "Pacific Gas & Electric Company": 2,
+        },
+        "CA Electricity Demand Forecast Zones": {
+            "LADWP Coastal": 0,
+            "Greater Bay Area": 1,
+        },
+        "CA Electric Balancing Authority Areas": {"BANC": 0, "CALISO": 1},
+    }
+    mock_boundaries.boundary_dict.return_value = boundary_dict
+
+    # Stations dataframe
+    mock_stations_df = pd.DataFrame(
+        {
+            "LON_X": [-120.0, -121.0],
+            "LAT_Y": [35.0, 36.0],
+            "station_name": ["Station1", "Station2"],
+        }
+    )
+
+    return mock_esm_catalog, mock_boundary_catalog, mock_boundaries, mock_stations_df
+
+
+@pytest.fixture
+def mock_data_catalog_and_objs() -> Tuple:
```

Comment:
Tuple technically isn't correct here, it should be `Generator`. The `yield` key word means the return type is always a generator object, not a tuple, or list.